### PR TITLE
[GOBBLIN-928]Craftsmanship cleaning and bumping up ORC version

### PR DESCRIPTION
**Pull request by autumnust**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- https://issues.apache.org/jira/browse/GOBBLIN-928

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by autumnust** [comment]
@sv2000  can you take a look? thanks

File: gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobLauncher.java [hunk 1]
```diff
@@ -381,7 +381,8 @@ public void launchJob(@Nullable JobListener jobListener)
   }
 
   /**
-   * Add a single {@link WorkUnit} (flattened).
+   * Add a single {@link WorkUnit} (flattened) to persisted storage so that worker could fetch that based on information
```
Comment (sv2000) [inline gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobLauncher.java]: Nit: "persisted" -> "persistent". "could" -> "can".

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/StateStoreBasedWatermarkStorage.java [hunk 2]
```diff
@@ -50,7 +50,12 @@
   public static final String WATERMARK_STORAGE_TYPE_KEY ="streaming.watermarkStateStore.type";
   public static final String WATERMARK_STORAGE_TYPE_DEFAULT ="zk";
   public static final String WATERMARK_STORAGE_CONFIG_PREFIX="streaming.watermarkStateStore.config.";
-  private static final String WATERMARK_STORAGE_PREFIX="streamingWatermarks:";
+
+  /**
+   * The prefix need to be fitting with different storage layer. e.g. some of characters won't be allowed in URI and
```
Comment (sv2000) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/StateStoreBasedWatermarkStorage.java]: A bit of word-smithing: "A watermark prefix that is compatible with different watermark storage implementations. As such, this prefix should not include any characters disallowed in a {@link URI}."

File: gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobLauncher.java (hunk 1 above)
Comment (autumnust) [reply gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobLauncher.java]: Thanks, addressed.

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/StateStoreBasedWatermarkStorage.java (hunk 2 above)
Comment (autumnust) [reply gobblin-runtime/src/main/java/org/apache/gobblin/runtime/StateStoreBasedWatermarkStorage.java]: Thanks, addressed.

**Pull request closed without merging**